### PR TITLE
Add `@ianvs/prettier-plugin-sort-imports`

### PR DESCRIPTION
**Pull request by jasikpark**
## Changes

- Adds `@ianvs/prettier-plugin-sort-imports`
- Formats the code w/ it + it will be formatted in ci via `pnpm run format`
- This plugin respects side effect imports unlike `organize-imports-cli` & I've had success w/ it. It's a fork of a trivago version that did not respect side effect imports.

continuing https://github.com/withastro/astro/pull/3013

cc @FredKSchott @IanVS

### CODE REVIEW

**Comment by jasikpark** [comment]
The cool thing is that it works w/ `prettier-plugin-astro` as well -- since the frontmatter uses the typescript parser 👀

**Comment by IanVS** [comment]
Some of these look backwards, it put the third-party imports _after_ local files, which isn't correct.  I can look into it.

**Comment by natemoo-re** [comment]
Nice! Can we also add [0a41422](https://github.com/withastro/astro/pull/3029/commits/0a414228ab68c1c3ab99014ef43bcb57f9234ae4) to [`.git-blame-ignore-revs`](https://github.com/withastro/astro/blob/main/.git-blame-ignore-revs)?

**Comment by jasikpark** [comment]
Sure! I was planning on doing that in a different PR, since I assume this will be squash merged & that git revision will no longer be accurate?

**Comment by jasikpark** [comment]
What do you think is the best course @natemoo-re 

a. Merge this PR sans git revision hash and create a second PR that is based on the squash commit of this PR
b. Rebase this branch so that it's pretty and can be merged via a regular merge as 3 different commits

**Comment by FredKSchott** [comment]
So what I mainly wanted to get out of #3013 was auto-removing unused imports, including TS types. Does this do that? Can you test to confirm? I can't tell based on the diff, since it adds more lines than it removes.

I actually care about sort order much less than that. fwiw, if you read my comment in #3013, the side-effect imports issue will solve itself in the next TypeScript patch, where you can just place it above the others in its own group, which VSCode will respect.

As long as this PR can auto-remove unused TS type imports AND unused named exports, then I'm pretty happy either way. Conceptually I like the idea of relying on TypeScript directly vs. a fork of an ESLint plugin, but I also wasn't super impressed with that TS CLI, so... 🤷

**Comment by IanVS** [comment]
No, the prettier plugin does not remove imports, and I wouldn't want to add that functionality to it either, as it is out-of-scope for its purpose and personally I would not want that behavior.  Since it sounds like that's what you're looking for, I'd recommend closing this PR and waiting for the typescript tool.

**Comment by jasikpark** [comment]
So this isn't an eslint plugin, it's a prettier plugin, so it runs with every `pnpm run format`; It doesn't do auto import removal, though from what I understand about typescript that can be a bit  iffy. 

I'd use `organize-imports-cli` if auto-removing imports is your primary aim rather than sorting imports.

**Comment by jasikpark** [comment]
I was working off your title 😅

**Comment by FredKSchott** [comment]
> So this isn't an eslint plugin, it's a prettier plugin, so it runs with every pnpm run format

my mistake! I misstyped 

> It doesn't do auto import removal, though from what I understand about typescript that can be a bit iffy.

It's always worked for me over 4+ years (through VSCode), but if you can point to anything that would be good reason to hold off I'd love to see it. 

But yea, removing unused imports and consistently formatting the remaining imports are the two things that I wanted to tackle. If everyone is good for waiting a few more days, then that original PR should be good to reopen to tackle both (and I can reword it to be more clear).
